scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/TypescriptCodeGenerationService.java

```diff
@@ -0,0 +1,164 @@
+
+  public static String generateTypescriptTypesFromMissionModel(final MissionModelTypes missionModelTypes) {
+    var indentLevel = 0;
+    final var activityTypeCodes = new ArrayList<ActivityTypeCode>();
+    for (final var activityType : missionModelTypes.activityTypes()) {
+      activityTypeCodes.add(getActivityTypeInformation(activityType));
+    }
+    var result = "/** Start Codegen */\n";
+    for (final var activityTypeCode : activityTypeCodes) {
+      result += "interface %s extends ActivityTemplate {}\n".formatted(activityTypeCode.activityName()).indent(indentLevel);
+    }
+    result += "export const ActivityTemplates = {\n".indent(indentLevel);
+    indentLevel += 2;
+    for (final var activityTypeCode : activityTypeCodes) {
+      result += String.format("%s: function %s(\n", activityTypeCode.activityName(), activityTypeCode.activityName()).indent(indentLevel);
+      indentLevel += 2;
+      result += "name: string\n".indent(indentLevel);
+      result += "args: {\n".indent(indentLevel);
+      indentLevel += 2;
+      for (final var parameterType : activityTypeCode.parameterTypes()) {
+        result += "%s: %s\n".formatted(parameterType.name(), ActivityParameterType.toString(parameterType.type())).indent(indentLevel);
+      }
+      indentLevel -= 2;
+      result += "}): %s {\n".formatted(activityTypeCode.activityName()).indent(indentLevel);
+      indentLevel += 2;
+      result += "return { name, activityType: '%s', args };\n".formatted(activityTypeCode.activityName()).indent(indentLevel);
+      indentLevel -= 2;
+      result += "},\n".indent(indentLevel);
+    }
+    result += "}\n/** End Codegen */";
+    return result;
+  }
+
+  private record ActivityTypeCode(String activityName, List<ActivityParameter> parameterTypes) {}
+  private record ActivityParameter(String name, ActivityParameterType type) {}
+  private sealed interface ActivityParameterType {
+    record TSString() implements ActivityParameterType {}
+    record TSDouble() implements ActivityParameterType {}
+    record TSBoolean() implements ActivityParameterType {}
+    record TSInt() implements ActivityParameterType {}
+    record TSDuration() implements ActivityParameterType {}
+    record TSArray(ActivityParameterType elementType) implements ActivityParameterType {}
+    record TSStruct(List<Pair<String, ActivityParameterType>> keysAndTypes) implements ActivityParameterType {}
+    record TSEnum(List<String> values) implements ActivityParameterType {}
+
+    /**
+     * Print this type out in one line.
+     */
+    static String toString(final ActivityParameterType type) {
+      if (type instanceof TSString) {
+        return "string";
+      } else if (type instanceof TSDouble) {
+        return "AST.Double";
+      } else if (type instanceof TSBoolean) {
+        return "boolean";
+      } else if (type instanceof TSInt) {
+        return "AST.Integer";
+      } else if (type instanceof TSDuration) {
+        return "AST.Duration";
```

Comment:
Latest force push adds the `AST.` prefix to `Integer` and `Duration`. @dyst5422 is there a more preferable way to refer to these types? I could import them all explicitly from `scheduler-ast.ts`...?

Reply:
We can supply these to the global typespace as just `Integer`, `Duration` etc and not have to prefix them, which I think is a better option